sql-plugin/src/main/spark330/scala/com/nvidia/spark/rapids/shims/GpuIntervalUtils.scala

```diff
@@ -165,7 +165,11 @@ object GpuIntervalUtils {
     withResource(castStringToDTInterval(cv, t)) { ret =>
       if(ret.getNullCount > cv.getNullCount) {
         throw new IllegalArgumentException("Cast string to day time interval failed, " +
-            "may be the format is invalid, range check failed or overflow")
+            "maybe the format is invalid, range check failed or overflow. " + 
+            "Do not support short mode, take DAY TO SECOND for example: " +  
+            "not support short mode: 100 10:30:40.999999, " +  
+            "only support the normal mode: "  +
+            "INTERVAL '100 10:30:40.999999' DAY TO SECOND")
```

Comment:
Thanks for the pointer,it is more clear to me now.  

So in the plugin doc we make the distinction between ANSI style vs HIVE style. 
I am also not sure why our doc focuses only on CSV whereas this seems to be a general string-to-interval casting issues.

```Python
>>> import pyspark.sql.functions as f
>>> import pyspark.sql.types as t
>>> df = spark.createDataFrame([("010101",)], ['n'])
>>> df1=spark.createDataFrame([('101 10:30:40.999999',),], 'a string').repartition(2)
>>> df2=spark.createDataFrame([("INTERVAL '101 10:30:40.999999' DAY TO SECOND",),], 'a string').repartition(2)
>>> df1.select(df1.a.cast(t.DayTimeIntervalType(t.DayTimeIntervalType.DAY, t.DayTimeIntervalType.SECOND))).collect()
...
Caused by: java.lang.IllegalArgumentException: Cast string to day time interval failed, may be the format is invalid, range check failed or overflow

>>> df2.select(df2.a.cast(t.DayTimeIntervalType(t.DayTimeIntervalType.DAY, t.DayTimeIntervalType.SECOND))).collect()

Row(a=datetime.timedelta(days=101, seconds=37840, microseconds=999999))]
```
In the upstream doc https://spark.apache.org/docs/latest/sql-ref-literals.html#interval-literal  the HIVE style is called multi-units syntax
>  The interval literal supports two syntaxes: ANSI syntax and multi-units syntax.

which is probably what we should adopt in spark-rapids doc as well. Regardless of that in the error message it is easier to focus on what we support instead of giving examples of what we do not support IMO

```suggestion
            "Maybe the format is invalid: only the ANSI syntax with the INTERVAL keyword is supported for day time interval type instances https://nvidia.github.io/spark-rapids/docs/compatibility.html#csv-ansi-day-time-interval. Maybe, the range / overflow check failed")
```